projects/core/src/auth/user-auth/facade/auth.service.ts

```diff
@@ -43,13 +43,15 @@ export class AuthService {
    * Check params in url and if there is an code/token then try to login with those.
    */
   async checkOAuthParamsInUrl(): Promise<void> {
-    try {
-      const tokenReceived = getLastValueSync(
-        this.oAuthLibWrapperService.events$.pipe(
-          filter((e) => e.type === 'token_received')
-        )
-      );
+    let tokenReceived: OAuthEvent | undefined;
```

Comment:
What do you think about adding a code comment explaining why we need this variable?
```ts
/**
  * The event 'token_received' is emitted, when `oAuthLibWrapperService.tryLogin()`
  * can obtain the token from the URL params (which means we've just returned from 
  * an external login page to Spartacus).
  * 
  * Please note that this event is not emitted when `oAuthLibWrapperService.tryLogin()`
  * can obtain the token from the storage (e.g. on refresh of the Spartacus page).
  */
let tokenReceived: OAuthEvent | undefined;
```